src/operator/numpy/np_broadcast_reduce_op_value.cc

```diff
@@ -320,6 +321,37 @@ NNVM_REGISTER_OP(_backward_np_average)
     return std::vector<ResourceRequest>{ResourceRequest::kTempSpace};
 });
 
+inline bool NumpyMedianType(const nnvm::NodeAttrs& attrs,
+                              std::vector<int> *in_attrs,
+                              std::vector<int> *out_attrs) {
+  CHECK_EQ(in_attrs->size(), 1U);
+  CHECK_EQ(out_attrs->size(), 1U);
+
+  TYPE_ASSIGN_CHECK(*in_attrs, 0, out_attrs->at(0));
```

Comment:
The type for output should `float32` by default (or floating number of other precisions), instead of being inferred from the input.